src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -44,32 +45,41 @@ object Ledger extends Logger {
     stateStorage: NodeStorage):
   ExecResult = {
     val blockchain = BlockchainImpl(storages)
-    val initialWorldStateProxy = InMemoryWorldStateProxy(storages, stateStorage,
+    val initialWorld = InMemoryWorldStateProxy(storages, stateStorage,
       blockchain.getBlockHeaderByHash(block.header.hash).map(_.stateRoot)
     )
-    block.body.transactionList.foldLeft[ExecResult](initialWorldStateProxy -> 0) {
-      case ((worldStateProxy, acumGas), stx) =>
-        val result: Either[String, PR] = for {
-          _ <- validateTransaction(stx, worldStateProxy, 0, block)
-          worldStateProxy1 = updateAccountBeforeExecution(stx, worldStateProxy)
-          result <- execute(stx, block.header, worldStateProxy1)
-        } yield result
-
-        result match {
-          case Right(theResult) =>
-            val gasUsed = stx.tx.gasLimit - theResult.gasRemaining
-            val refundGasFn = {
-              if (theResult.gasRemaining > 0) (refundGas _).curried(Address(block.header.beneficiary))(gasUsed)
-              else identity[InMemoryWorldStateProxy] _
-            }
-            val payBeneficiariesFn = (payForGasUsedToBeneficiary _).curried(stx)
-            val deleteAccountsFn = (deleteAccounts _).curried(theResult.addressesToDelete)
-            (refundGasFn andThen payBeneficiariesFn andThen deleteAccountsFn) (theResult.world) -> (gasUsed + acumGas)
-          case Left(error) => throw new RuntimeException(error)
+    block.body.transactionList.foldLeft[ExecResult](initialWorld -> 0) {
+      case ((world, acumGas), stx) =>
+        //TODO: change 0 -> acumGas ?
```

Comment:
Should this be changed in this PR? I think you are right about having to do this change.